go-controller/pkg/ovn/host_subnet_allocator.go

```diff
@@ -0,0 +1,182 @@
+package ovn
+
+import (
+	"fmt"
+	"net"
+
+	kapi "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	utilnet "k8s.io/utils/net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/metrics"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/subnetallocator"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+type hostSubnetAllocator struct {
+	// Don't inherit from BaseSubnetAllocator to ensure users of
+	// hostSubnetAllocator can't directly call the underlying methods
+	base *subnetallocator.BaseSubnetAllocator
+}
+
+func NewHostSubnetAllocator() *hostSubnetAllocator {
+	return &hostSubnetAllocator{
+		base: subnetallocator.NewSubnetAllocator(),
+	}
+}
+
+func (sna *hostSubnetAllocator) InitRanges(subnets []config.CIDRNetworkEntry) error {
+	for _, entry := range subnets {
+		if err := sna.base.AddNetworkRange(entry.CIDR, entry.HostSubnetLength); err != nil {
+			return err
+		}
+		klog.V(5).Infof("Added network range %s to host subnet allocator", entry.CIDR)
+	}
+
+	// update metrics for host subnets
+	v4count, _, v6count, _ := sna.base.Usage()
+	metrics.RecordSubnetCount(float64(v4count), float64(v6count))
+	return nil
+}
+
+// We only deal with cleaning up nodes that shouldn't exist here, since
```

Comment:
I think this came accidently from:
```
// We only deal with cleaning up nodes that shouldn't exist here, since
// watchNodes() will be called for all existing nodes at startup anyway.
// Note that this list will include the 'join' cluster switch, which we
// do not want to delete.
func (oc *Controller) syncNodesRetriable(nodes []interface{}) error {
``` 
:)

Reply:
Good catch, removed this and added a real docstring for MarkSubnetsAllocated()